torch/quantization/fx/_equalize.py

```diff
@@ -0,0 +1,231 @@
+import torch
+import torch.nn as nn
+from torch.quantization.observer import MinMaxObserver, PerChannelMinMaxObserver
+
+import warnings
+
+
+class _InputEqualizationObserver(nn.Module):
+    r"""Observer for tracking the running min/max values of input columns, and
+    computing the quantization parameters for the overall min/max input values.
+
+    Args:
+        dtype: Quantized data type
+        qscheme: Quantization scheme
+        quant_min: Minimum quantization value. If unspecified, it will
+            follow the 8-bit setup.
+        quant_max: Maximum quantization value. If unspecified, it will
+            follow the 8-bit setup.
+        output_obs: For the user to specify what kind of output observer they
+            would like to use
+
+    The running minimum/maximum :math:`x_\text{min/max}` are computed in the
+    same way as :class:`~torch.quantization.observer.PerChannelMinMaxObserver`,
+    with the difference that the running min/max values are stored per column.
+
+    The qparams are calculated by multiplying the min/max input column values
+    with the equalization scale, reducing to find the global min/max input
+    values, and then calculating in the same way as in
+    :class:`~torch.quantization.observer.MinMaxObserver`
+
+    .. note:: If the running minimum equals to the running maximum, the scales
+              and zero_points are set to 1.0 and 0.
+    """
+
+    def __init__(self, dtype=torch.quint8, qscheme=torch.per_tensor_affine,
+                 quant_min=None, quant_max=None, output_obs=None,
+                 factory_kwargs=None) -> None:
+        super(_InputEqualizationObserver, self).__init__()
+
+        if qscheme not in {torch.per_tensor_affine, torch.per_tensor_symmetric}:
+            raise TypeError("Input qscheme must be per-tensor")
+
+        self.input_obs = PerChannelMinMaxObserver(ch_axis=1, dtype=dtype,
+                                                  qscheme=qscheme,
+                                                  quant_min=quant_min,
+                                                  quant_max=quant_max,
+                                                  factory_kwargs=factory_kwargs)
+
+        if output_obs is None:
```

Comment:
actually I think this should probably be a constructor as well? otherwise all the Tensors in the model will use the same output observer if user customizes it.